### PR TITLE
stats: move regex pattern to companion object

### PR DESCRIPTION
**Pull request by jingwei99**
Description: Re-use regex pattern across all `Element` instances, since `Pattern.compile()` comes with performance cost.
Risk Level: Low
Testing: Unit test

Signed-off-by: Jingwei Hao <jingwei.hao@gmail.com>

### CODE REVIEW

File: library/kotlin/test/io/envoyproxy/envoymobile/ElementTest.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  @Test
+  fun `creates new element as expected`() {
+    // Should just work
+    Element("foo")
+  }
+
+  @Test(expected = IllegalArgumentException::class)
+  fun `throw exception when element name is rejected`() {
+    // Should just work
```
Comment (buildbreaker) [inline library/kotlin/test/io/envoyproxy/envoymobile/ElementTest.kt:15]:
```suggestion
    // Should throw
```?

**Review by junr03 — COMMENTED** [review]
lgtm, @buildbreaker on the kotlin pattern.

Do we do a similar check in swift?

**Comment by buildbreaker** [comment]
@jingwei99 Could we actually get some more consistency between kotlin and swift: https://github.com/lyft/envoy-mobile/blob/46503043b9d1d539bdbe45b647f17f18902f46e2/library/swift/src/stats/Element.swift#L3?

**Comment by jingwei99** [comment]
> @jingwei99 Could we actually get some more consistency between kotlin and swift:
> 
> https://github.com/lyft/envoy-mobile/blob/46503043b9d1d539bdbe45b647f17f18902f46e2/library/swift/src/stats/Element.swift#L3
> 
> ?

it is the same check on the pattern, right? or you mean something different?

**Comment by buildbreaker** [comment]
I meant let's try to get the structure and variable names the same too

**Comment by buildbreaker** [comment]
Sorry to clarify what I meant was to mirror the `value` field name and internal visibility from swift

**Review by buildbreaker — APPROVED** [review]
Thanks!

**Pull request merged**